### PR TITLE
Updating IPU SchNet example for PopTorch with SDK 3.0

### PR DESCRIPTION
**Pull request by hatemhelal**

### CODE REVIEW

File: get-started/PyG-SchNetGNN.ipynb [hunk 1]
```diff
@@ -41,19 +41,30 @@
...
+    "from tqdm import tqdm\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import seaborn as sns\n",
+    "sns.set_theme()\n",
+    "\n",
+    "executable_cache_dir = os.getenv(\"POPLAR_EXECUTABLE_CACHE_DIR\", \"/tmp/exe_cache/\")\n",
+    "dataset_directory = os.getenv(\"DATASET_DIR\", 'data')\n",
+    "number_of_ipus = os.getenv(\"NUM_AVAILABLE_IPU\", 16)\n",
+    "number_of_ipus = min(number_of_ipus, 16) # QM9 is too small to benefit from additional scaling"
```
Comment (payoto) [inline get-started/PyG-SchNetGNN.ipynb:67]:
```suggestion
    "number_of_ipus = os.getenv(\"NUM_AVAILABLE_IPU\", \"16\")\n",
    "number_of_ipus = min(int(number_of_ipus), 16) # QM9 is too small to benefit from additional scaling"
```

File: get-started/PyG-SchNetGNN.ipynb [hunk 2]
```diff
@@ -40,20 +40,45 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%matplotlib inline\n",
+    "import os\n",
```
Comment (s-maddrellmander) [inline get-started/PyG-SchNetGNN.ipynb:43]: Could we have a `-q` on the pip install above?

File: get-started/PyG-SchNetGNN.ipynb [hunk 3]
```diff
@@ -123,32 +148,68 @@
...
-    "datum = dataset[0]\n",
+    "datum = dataset[123244]\n",
     "datum, datum.z, datum.pos, datum.y[:, 4]"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can apply a transform to the QM9 dataset that selects the properties we need for training SchNet"
```
Comment (s-maddrellmander) [inline get-started/PyG-SchNetGNN.ipynb:186]:
```suggestion
    "Using a transform to the QM9 dataset we can select the properties we need for training SchNet and convert them to a PyG Data object."
```
Sounded like we were doing more transforming of the data than just selecting + turning into the correct type

**Review by s-maddrellmander — APPROVED** [review]
Looks really good to me, a few small comments / generalisations, but they can mostly be put onto a backlog if you prefer. 

And a couple of questions for my understanding. 
Otherwise looks good!

File: get-started/PyG-SchNetGNN.ipynb (hunk 3 above)
Comment (hatemhelal) [reply get-started/PyG-SchNetGNN.ipynb:186]: thanks your version reads much easier

File: get-started/pyg_schnet_util.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    """
+    Creates a data loader for graph datasets
+    Applies the mini-batching method of concatenating multiple graphs into a 
+    single graph with multiple disconnected subgraphs. See:
+    https://pytorch-geometric.readthedocs.io/en/2.0.2/notes/batching.html
+    
+    Automatically adds a padding graph to fill up each mini-batch up to contain
+    max_nodes_per_graph * (batch_size-1) nodes.
+    """
+    keys = ("z", "pos", "batch", "y")
```
[the remark this answers is not in the document]
Comment (hatemhelal) [reply get-started/pyg_schnet_util.py:169]: Sure, I made it an argument with a default value for SchNet to keep the notebook code simple.

Also added an assertion if the padding graph has a negative number of nodes.

File: get-started/PyG-SchNetGNN.ipynb [hunk 5]
```diff
@@ -1121,17 +928,16 @@
...
-    "device_iterations = 6\n",
-    "gradient_accumulation = 2\n",
-    "num_dataloader_workers = 16\n",
-    "learning_rate = 1e-3\n",
-    "num_epochs = 20"
+    "batch_size = 8\n",
+    "replication_factor = num_ipus\n",
+    "device_iterations = 32\n",
+    "gradient_accumulation = 16 // num_ipus\n",
+    "learning_rate = 1e-4\n",
```
Comment (payoto) [inline get-started/PyG-SchNetGNN.ipynb:939]: Attempt a multiplication by 16

```suggestion
    "gradient_accumulation = 256 // num_ipus\n",
    "learning_rate = 2e-3\n",
```

File: get-started/PyG-SchNetGNN.ipynb [hunk 6]
```diff
@@ -4,9 +4,9 @@
    "cell_type": "markdown",
```
Comment (payoto) [inline get-started/PyG-SchNetGNN.ipynb:4]: <p>Test comment</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/Graphcore-Pytorch-fork/pull/8/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>
